src/main/java/com/Model/Patient.java

```diff
@@ -37,4 +37,18 @@ public void addAttendEvents(Event event){
         return this.attendedEvents;
     }
 
+    /**
+     *Return true if two patients have the same Id
+     */
+    @Override
+    public boolean equals (Object o){
+        if (o instanceof Patient){
+            Patient p = (Patient) o;
+            if (p.Id == this.Id){
```

Comment:
I honestly don't know about this: should we just always compare the id's and not bother with the other fields? If so, why not just do it everywhere? If not, then why here?

Reply:
Because for events, if two users input two events that have the same place, name and date, they can be treated as the same event. 
Same for the Place, if two places share the same address, longitude and latitude, they are the same place. 
For patients, two patients may attend exactly same events. So I just distinguish them with ID.